### DISCUSSION TITLE
Help needed to integrate FastAPI-users in my app under dev. SQLAlchemy

### DISCUSSION THREAD

**Discussion by Enginiosys** [fastapi-users/fastapi-users#872] [Q&A]
Hi,

I am desperately looking for help :) I went trough all the docs but I am still stuck.
I created a "big" application with fastApi including a router and 'sub apps' as recommended in fast-api tutorial.
I am using SQL alchemy and alembic for my migrations.

My objective is now to move from a simple user with no act to a secured authentication. using Bearer for transport and Database strategy.
for each my sub apps I am using the 'official structure' with a crud, schemas files. So when I downloaded the fatsapi-users project I am completely lost.

concretely, in my model.py file, I upgraded my user model to the following to be able to migrate it and use the model in my app.

`
class User(Base, SQLAlchemyBaseUserTable):

    joined_date = Column(DateTime)
    last_connection_date = Column(DateTime)
    contact_number = Column(String, unique=True)
    language = Column(String, default="EN")
    used_referral_code = Column(String, default="EN")
    own_referral_code = Column(String, unique=True, nullable=True)
    country = Column(String)
    two_fa_is_enabled = Column(Boolean, default=False)
    two_fa_code = Column(String, unique=True, nullable=True)
`

and I also created the following : 

`
class OAuth(Base, SQLAlchemyBaseOAuthAccountTable):
    pass
`

Also I 

So my question : 
-  what about the SQLAlchemyUserDatabase(BaseUserDatabase[UD]) function ? 

- When trying to migrate, I have the following error : name 'fastapi_users_db_sqlalchemy' is not defined.. I suppose I am not approaching the integration the correct way.

- Also is it possible to have a short guidance on the different steps to integrate it properly please?
I may be wrong, but for now What I did is : 
1- started with the examples.sqlachemy project folder.
2- I moved the app.py content to my main.py files ( the one calling the different app etc)
3- left the models.py files as is for now. (If I understand well, this is what is name schemas.py in big application in the official docs)

But I have no idea on how to integrate the models properly and how to use the db.py and users.py files properly?

Any help/guidance will be really appreciated.

thank you in advance for your time guys  !

Yassine

**Comment by frankie567** [comment]
I suggest you to read the documentation, not just looking at the examples: https://fastapi-users.github.io/fastapi-users/configuration/overview/

It'll help you understand how everything fit together.

**Reply by Enginiosys** [reply]
I already did. I went trough everything already.
